### ISSUE TITLE
Vengeful Spirit Aghanim's Shard and refreshing Magic Missile interaction

### ISSUE THREAD

**Issue by keepertheslayer** [ValveSoftware/Dota2-Gameplay#11925]
### Ability name

Magic Missile

### Description

TL;DR When the Venge is casting two Magic Missiles in quick succession (e.g. using Refresher Orb), Shard's ability work incorrectly or doesn't work at all.

Venge's Aghanim's Shard description says: "Bounces once on an enemy, prioritizing heroes, within 75% of Magic Missile's current cast range from the first target".

So, no matter how exactly the spells are being cast, every time for every cast Magic Missile hit order should be: Primary target (initial hit) => Secondary target (bounce hit), which is not what's happening.

As far as I can see there are three possible variations of the spellcasting order/proximity:

1. Magic Missile 1 cast => Target 1 hit (initial) => Magic Missile 2 cast => Target 2 hit (bounce from the first cast) => ...
2. Magic Missile 1 cast => Magic Missile 2 cast => Target 1 hit (initial) => ...
3. Magic Missile 1 cast => Target 1 hit (initial) => Target 2 hit (bounce from the first cast) => Magic Missile 2 cast => ...

Using cases from above:
1. When the second Magic Missile (MM2) is cast between the initial hit from the first Magic Missile (MM1) and its bounce, then it looks like the bounce for the MM2 is being produced by MM1 bounce hit and not by MM2 initial hit, like it should be.

If you're casting spells on different targets, the first one may receive the damage and stun twice even before your second spell hits its target.

Magic Missile 1 cast => Target 1 hit (initial) => Magic Missile 2 cast => Target 2 hit (bounce from the first cast) => Target 1 hit (bounce from bounce?) => Target 2 hit (initial)

It gets even better when you're casting both spells on the same target - second spell's bounce procs on the first spell's bounce and as a result your primary target will be hit three times and secondary target only once.

2. When the second Magic Missile (MM2) is cast before the initial hit from the first Magic Missile (MM1), then there are no bounce at all for the MM2.

Magic Missile 1 cast => Magic Missile 2 cast => Target 1 hit (initial) => Target 2 hit (bounce from the first cast) => Target 2 hit (initial)

If casting on the same target, then primary target will be hit two times and secondary target once.

3. When the second Magic Missile (MM2) is cast after the first Magic Missile (MM1) hits both initial target and the bounce, then MM2 works fine and both targets will be hit with every Magic Missile cast.

Tested all those cases in lobby and results are the same.

Combat log screenshot:
![combatlog](https://github.com/ValveSoftware/Dota2-Gameplay/assets/46645566/3d5baed8-a905-4e4c-b651-7dbbf1c0fd53)

Videos:
https://www.youtube.com/watch?v=zPB-7KP99fM
https://www.youtube.com/watch?v=NjnKwR8AYn4

### Example Match ID (and possibly Timestamp)

No Match ID, all tests were done in lobby and demo mode
